### DISCUSSION TITLE
Typing effect in React with typed.js library inconsistent in Safari browser.

### DISCUSSION THREAD

**Discussion by asdacosta** [community/community#131096] [Programming Help]
### Body

I am using typed.js to animate input placeholders. On Google Chrome, the typing effect behaves normal even when I resize the screen. I used the typed library in a useEffect hook with an empty array list as shown below.

However, on Safari browser on iphone (specifically tried with iphone 7 plus), a quick inconsistent type occurs when a new loop of the stirngs is about to begin. 
It behaves as if there is an if-condition specified in useEffect and a variable is listed in the array list to cause a re-render (I tried to use a condition and a dependency in Chrome and it had similar effect). 

So basically the typing effect in the Safari behaves normal but when the loop ends and it is about to start again, it behaves inconsistently. 
However, when I set the loop to false, I don't experience the inconsistency.

I have tried customizing with it's inbuilt functions but no progress.
I would like to understand the problem and possible solution.

This is the code:

```
 const typeFieldPlaceholder = (ref, placeholder) => {
  const typed = new Typed(ref.current, {
   strings: placeholder,
   typeSpeed: 40,
   backSpeed: 20,
   backDelay: 1200,
   attr: "placeholder",
   loop: true,
   smartBackspace: true,
   showCursor: false,
  });
  return typed;
 };

 useEffect(() => {
  const typed = typeFieldPlaceholder(typingNameRef, placeholders.names);
  return () => {
   typed.destroy();
  };
 }, []);
```

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Himanshu-Khadka** [comment]
It appears that the inconsistency you are experiencing in Safari might be due to the way the browser handles animations and re-renders. One possible solution is to ensure that Typed.js is initialized only once and to manage the state more effectively. You can try the following adjustments to your code:

1. **Refactor the Initialization**: Ensure that the `Typed` instance is created only once by using a `useRef` hook.

2. **Manage Component Lifecycle**: Make sure the component lifecycle is managed properly to prevent unnecessary re-renders.

Here is an example of how you could refactor your code:

```javascript
import React, { useEffect, useRef } from 'react';
import Typed from 'typed.js';

const YourComponent = () => {
  const typingNameRef = useRef(null);
  const typedInstance = useRef(null);

  const placeholders = {
    names: ["Placeholder 1", "Placeholder 2", "Placeholder 3"]
  };

  useEffect(() => {
    if (typingNameRef.current) {
      typedInstance.current = new Typed(typingNameRef.current, {
        strings: placeholders.names,
        typeSpeed: 40,
        backSpeed: 20,
        backDelay: 1200,
        attr: "placeholder",
        loop: true,
        smartBackspace: true,
        showCursor: false,
      });
    }

    return () => {
      if (typedInstance.current) {
        typedInstance.current.destroy();
      }
    };
  }, []);

  return (
    <input ref={typingNameRef} />
  );
};

export default YourComponent;
```

By using `useRef` for both the DOM element and the `Typed` instance, you ensure that the initialization happens only once, which may help in avoiding the inconsistency in Safari.

If the issue persists, consider checking for any specific CSS or other JavaScript interactions that might be affecting the re-rendering or animation in Safari. Additionally, you might want to test the latest version of Typed.js or report the issue to their repository for further assistance.

**Reply by asdacosta** [reply]
It is showing positive outcomes in Chrome and might likely work in Safari. 
I decided to remove the loop and trigger the typing effect only on every inView, which is the method I prefer now. So I don’t want to set a new deployment just to check if it works on Safari again and to later revert. 
I’ll try soon in my next project and give feedback when it works. 

Thank you @Himanshu-Khadka

**Comment by danukaji-M** [comment]


**Reply by ebndev** [reply]
Hi @danukaji-M, thanks for participating in GitHub Discussions!

Please note: we’ve clarified our stance on using generative AI tools like ChatGPT within our Community via [this announcement](https://github.com/orgs/community/discussions/69006). Please review the guidelines to ensure your post meets them as failure to adhere to those rules can result in action taken by our moderator team.  

You can read our updated [Code of Conduct](https://github.com/community/community/blob/baa5cd9020ef3f2bbf86dca1191d9c90f9b0822d/CODE_OF_CONDUCT.md#reasonable-use-of-ai-generated-content) and the [announcement](https://github.com/orgs/community/discussions/69006) for more details. 

Please feel encouraged to continue engaging in Discussions as your authentic self though. We appreciate you being a part of the Community 🌟
